common-libs/common/test/unit/utils/objectUtil.test.ts

```diff
@@ -0,0 +1,62 @@
+import { buildObjectSkeletonFromPaths, injectFieldForAllParentRoots } from '../../../src/utils/objectUtil'
+import { expect } from 'chai'
+
+describe('objectUtil', () => {
+  describe('#buildObjectSkeletonFromPaths', () => {
+    it('should build object that represents paths to fields', () => {
+      const paths = ['address/buildingNumber', 'address/country', 'id/age', 'id/name/firstName', 'date']
+
+      const representingPathsAsObject = buildObjectSkeletonFromPaths(paths)
+
+      expect(representingPathsAsObject).to.deep.equal({
+        date: {},
+        address: {
+          buildingNumber: {},
+          country: {},
+        },
+        id: {
+          age: {},
+          name: {
+            firstName: {},
+          },
+        },
+      })
+    })
+  })
+
+  describe('#injectFieldForAllParentRoots', () => {
+    it('should inject @explicit = true for all parent levels of object', () => {
+      const target = {
+        address: {
+          buildingNumber: {},
+          country: {},
+        },
+        id: {
+          age: {},
+          name: {
+            firstName: {},
+          },
+        },
+      }
+
+      const result = injectFieldForAllParentRoots(target, '@explicit', true)
+
+      expect(result).to.deep.equal({
+        address: {
+          buildingNumber: {},
+          country: {},
+          '@explicit': true,
```

Comment:
Is there any documentation reference for why `address` should contain `@explicit` but `country` should not (even though it is an object)?

Reply:
If a field is a parent field (e.g. contains nested) it should have '@explicit': true flag to indicate only specified fields should be selected. if a field is not a parent -> then it doesn’t make sense to use '@explicit' flag.
I left some links to examples why we use this syntax below, also README.md of jsonld-signatures-bbs https://github.com/mattrglobal/jsonld-signatures-bbs